internal/logger/logger.go

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright (c) F5, Inc.
+ *
+ * This source code is licensed under the Apache License, Version 2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package logger
+
+import (
+	"io"
+	"log/slog"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/nginx/agent/v3/internal/config"
+)
+
+const (
+	defaultLogFile = "agent.log"
+)
+
+var logLevels = map[string]slog.Level{
+	"debug": slog.LevelDebug,
+	"info":  slog.LevelInfo,
+	"warn":  slog.LevelWarn,
+	"error": slog.LevelError,
+}
+
+func New(params config.Log) *slog.Logger {
+	handler := slog.NewTextHandler(
+		getLogWriter(params.Path),
+		&slog.HandlerOptions{
+			Level: getLogLevel(params.Level),
+		},
+	)
+
+	return slog.New(handler)
+}
+
+func getLogLevel(level string) slog.Level {
+	if level == "" {
+		return slog.LevelInfo
+	}
+	return logLevels[strings.ToLower(level)]
```

Comment:
What about if the log level is not found from the `logLevels` map?

Reply:
It returns level info by default if not found